granite: 5.2.1 -> 5.2.2

worldofpeace opened the pull request:
###### Motivation for this change
Closes #52882

Usual update but with stuff that I've already done to granite in https://github.com/NixOS/nixpkgs/pull/48637 that should be on master.
Most importantly this should fix certain elementary third-party applications from breaking.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

worldofpeace committed:
granite: 5.2.1 -> 5.2.2

worldofpeace commented:
@GrahamcOfBorg build deepin.deepin-wm

jtojnar commented on pkgs/development/libraries/granite/default.nix:
```diff
@@ -1,38 +1,53 @@
-    gnome3.libgee
     gtk3
+    hicolor-icon-theme
+    gnome3.libgee
```
Can be just `libgee` now.

worldofpeace replied:
That change isn't on master yet.

jtojnar commented on pkgs/development/libraries/granite/default.nix:
```diff
@@ -1,38 +1,53 @@
+      sha256 = "0cdp9ny6fj1lpcirab641p1qn1rbsvnsaa03hnr6zsdpim96jlvs";
+    })
+    # Resolve the circular dependency between granite and the datetime wingpanel indicator
+    ./02-datetime-clock-format-gsettings.patch
```
Please add a link to the patch source.

worldofpeace committed:
granite: various cleanups

This is just stuff that's in #48637 already that I wanted to sync.

In detail it's:

  - Patching in the use of meson

  - Add wrapGAppsHook and hicolor-icon-theme
    There's a demo application and granite
     provides some generic icons.

  - Included patches so third party apps still work

worldofpeace commented:
Thanks @jtojnar
